src/solver/optimisation/adequacy_patch_csr/set_problem_cost_function.cpp

```diff
@@ -37,17 +37,19 @@
 
 namespace
 {
-double calculateQuadraticCost(const PROBLEME_HEBDO* problemeHebdo, int hour, int area)
+
+using namespace Antares::Data::AdequacyPatch;
+
+double calculateQuadraticCost(const PROBLEME_HEBDO* problemeHebdo, const AdqPatchPTO priceTakingOrder, int hour, int area)
 {
     double priceTakingOrders = 0.0; // PTO
-    if (problemeHebdo->adqPatchParams->PriceTakingOrder == Data::AdequacyPatch::AdqPatchPTO::isLoad)
+    if (priceTakingOrder == Data::AdequacyPatch::AdqPatchPTO::isLoad)
```

Comment:
`switch..case` ?

In any case, add `using Data::AdequacyPatch` at the top of this function

Reply:
Done